test/distributed/org/apache/cassandra/distributed/fuzz/HistoryBuilderIntegrationTest.java

```diff
@@ -0,0 +1,213 @@
+             });
+    }
+
+    public void test(Consumer<HistoryBuilder> test)
+    {
+        Supplier<SchemaSpec> gen = SchemaGenerators.progression(SchemaGenerators.DEFAULT_SWITCH_AFTER);
+        for (int i = 0; i < SchemaGenerators.DEFAULT_RUNS; i++)
+        {
+            Configuration config = configuration(i, gen.get());
+            Run run = config.createRun();
+            cluster.schemaChange("DROP KEYSPACE IF EXISTS harry");
+            cluster.schemaChange("CREATE KEYSPACE harry WITH replication = {'class': 'SimpleStrategy', 'replication_factor': " + RF + "};");
+            run.sut.schemaChange(run.schemaSpec.compile().cql());
+            System.out.println("run.schemaSpec = " + run.schemaSpec.compile().cql());
+            HistoryBuilder history = new HistoryBuilder(run);
+
+            Set<Long> pds = new HashSet<>();
+            run.tracker.onLtsStarted((long lts) -> {
+                pds.add(run.pdSelector.pd(lts, run.schemaSpec));
+            });
+
+            ReplayingVisitor visitor = history.visitor(run);
+            test.accept(history);
+            visitor.replayAll();
+
+            Model model = new QuiescentChecker(run, new Reconciler(run,
+                                                                   history::visitor));
+            QueryGenerator.TypedQueryGenerator queryGenerator = new QueryGenerator.TypedQueryGenerator(run);
+            Assert.assertFalse(pds.isEmpty());
+            for (Long pd : pds)
+            {
+                model.validate(Query.selectPartition(run.schemaSpec, pd,false));
+
+                int lts = new Random().nextInt((int) run.clock.peek());
+                for (int k = 0; k < 3; k++)
+                    queryGenerator.inflate(lts, k);
+            }
+        }
+    }
+
+    Configuration.ModelConfiguration modelConfiguration()
+    {
+        return new Configuration.QuiescentCheckerConfig();
+    }
+
+    public static class Counts
+    {
+        boolean randomOrder;
+        boolean simultaneously;
+        int partitionDeletion;
+        int update;
+        int insert;
+        int delete;
+        int rangeDelete;
+        int sliceDelete;
+        int columnDelete;
+
+        public void apply(HistoryBuilder sut)
+        {
+            HistoryBuilder.PartitionBuilder builder = sut.nextPartition();
+            if (randomOrder)
+                builder.randomOrder();
+            else
+                builder.strictOrder();
+
+            if (simultaneously)
+                builder.simultaneously();
+            else
+                builder.sequentially();
+
+            builder.deletes(delete);
+            builder.inserts(insert);
+            builder.updates(update);
+            builder.partitionDeletions(partitionDeletion);
+            builder.rangeDeletes(rangeDelete);
+            builder.sliceDeletes(sliceDelete);
+            builder.columnDeletes(columnDelete);
+
+            builder.finish();
+        }
+        boolean allDone()
```

Comment:
unused

Reply:
removed from Cassandra test suite, since it is present in harry core test suite